### PR TITLE
Add snippets for the DateInterval type

### PR DESCRIPTION
**Pull request by carlosschults**
References #918.

### CODE REVIEW

File: src/NodaTime.Demo/DateIntervalDemo.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+﻿using NUnit.Framework;
```
Comment (jskeet) [inline src/NodaTime.Demo/DateIntervalDemo.cs]: Please add the copyright notice.

File: src/NodaTime.Demo/DateIntervalDemo.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+            Assert.AreEqual(365, interval.Length);
+            Assert.AreEqual("[2017-01-01, 2017-12-31]", interval.ToString());
+            Assert.AreEqual(start, interval.Start);
+            Assert.AreEqual(end, interval.End);
+            Assert.AreEqual(calendar, interval.Calendar);
+        }
+
+        [Test]
+        public void Intersection()
```
Comment (jskeet) [inline src/NodaTime.Demo/DateIntervalDemo.cs]: We can only handle a single snippet per method - I'd suggest that an intersection between "January-August" and "June-December" would be good.

File: src/NodaTime.Demo/DateIntervalDemo.cs [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            LocalDate start = new LocalDate(2017, 1, 1);
+            LocalDate end = new LocalDate(2017, 12, 31);
+
+            DateInterval interval = Snippet.For(new DateInterval(start, end));
+
+            Assert.False(interval.Contains(new LocalDate(1990, 6, 26)));
+        }
+
+        [Test]
```
Comment (jskeet) [inline src/NodaTime.Demo/DateIntervalDemo.cs]: This is the snippet I'd use for Contains(DateInterval)

File: src/NodaTime.Demo/DateIntervalDemo.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            DateInterval interval = Snippet.For(new DateInterval(start, end));
+            DateInterval june2018 = new DateInterval(
+                new LocalDate(2018, 6, 1),
+                new LocalDate(2018, 6, 30));
+
+            Assert.AreEqual(null, interval.Intersection(june2018));
+        }
+
+        [Test]
+        public void Contains()
```
Comment (jskeet) [inline src/NodaTime.Demo/DateIntervalDemo.cs]: This is the snippet I'd use for Contains(LocalDate)

**Comment by carlosschults** [comment]
Any more corrections needed here? @jskeet

**Comment by jskeet** [comment]
Will check shortly, thanks - hadn't seen the changes.

**Comment by jskeet** [comment]
(Otherwise fine though!)

**Comment by jskeet** [comment]
Hmm... This now seems to have a bunch of other commits in. I can't look now, but I'll see what I can do tonight.

**Comment by jskeet** [comment]
Okay, rebased and squashed everything down to one commit. I've removed a Snippet.For call; I'll merge on green and we can check how things look.

(Actually, I'm going to change the Assert.True calls into Assert.Equal calls for now, as that'll make the snippet output work better. I need to improve the snippet handling, basically...)

**Pull request merged**